plugin/src/main/java/org/opensearch/ml/task/MLPredictTaskRunner.java

```diff
@@ -185,6 +198,83 @@ protected void executeTask(MLPredictionTaskRequest request, ActionListener<MLTas
         }
     }
 
+    private void remoteInference(String modelId, MLTask mlTask, MLInput mlInput, ActionListener<MLTaskResponse> listener) {
```

Comment:
I think this private method can be abstracted into a template class, which can be directly reused by other third-party applications performing remote inference operations.